x-pack/plugins/fleet/server/services/setup.ts

```diff
@@ -33,7 +24,6 @@ import { awaitIfFleetServerSetupPending } from './fleet_server';
 export interface SetupStatus {
   isInitialized: boolean;
   preconfigurationError: { name: string; message: string } | undefined;
-  nonFatalPackageUpgradeErrors: DefaultPackagesInstallationError[];
```

Comment:
Is this undoing my work from https://github.com/elastic/kibana/pull/97404 ? Is this intentional?
Cc @ruflin

Reply:
Not intentional, I just wasn't immediately sure how to deal with the merge conflict. I'll get this back in.

Reply:
I think this actually mostly duplicates the functionality of `preconfigurationError`. I'll reconcile the differences and streamline.

Reply:
As of the latest commit, running the test scenario from #97404 results in a `preconfigurationError` of:

<img width="362" alt="Screen Shot 2021-04-19 at 2 20 39 PM" src="https://user-images.githubusercontent.com/1445834/115291748-bb634980-a11a-11eb-9d70-436a270aa8e7.png">

I can tweak this functionality as needed.

Reply:
I'm not sure about getting rid of `nonFatalPackageUpgradeErrors` and `DefaultPackagesInstallationError`. I like the explicitness of that interface wrt to package installation and various install types/ops: install/upgrade/rollback. package install handling is a sensitive area that we will want to keep iterating and improving on so I would like to keep the error granularity here.

in contrast, `preconfigurationError` can be related to package and agent policy errors.

my suggestion, what do you think?:

```ts
// export generic `nonFatalErrors` property
export interface SetupStatus {
  isInitialized: boolean;
  nonFatalErrors?: Array<PreconfigurationError |  DefaultPackagesInstallationError>;
}

// add back `DefaultPackagesInstallationError` and return it where appropriate
export interface DefaultPackagesInstallationError {
  installType: InstallType;
  error: Error;
}

// define `PreconfigurationError`
export interface PreconfigurationError {
  package?: { name: string; version: string; };
  agentPolicy?: { name: string; }
  error: Error;
}
```

Reply:
Implemented this as best I could. Currently all properties besides `error` go unused on both error types, but the data is there at least.